drivers/orbit-bhyve-sprinkler-timer.groovy

```diff
@@ -84,18 +86,29 @@ def uninstalled() {
     }
 }
 
+def on() {
+open()
+}
+
+def off() {
+close()
+}
+
 def open() {
     def runTime = presetRunTime ?: device.latestValue('preset_runtime') ?: 10
     parent.sendRequest('open', parent.getOrbitDeviceIdFromDNI(device.deviceNetworkId), device.latestValue('station'), runTime)
+    sendEvent(name: "switch", value: "on")
```

Comment:
I'd be concerned with putting this here. When open() is called all it means is we intend to send the command, not that it actually opened. The websocket might be closed, the hub might be offline, the valve might be completely turned off (no batteries). In all those cases, the valve would never actually open. That's why all the state changes are handled either in the poll in the app (updateTiles) and in the websocket handler in the driver (parse). The way I have it more closely models how Zwave and Zigbee drivers work where the driver state will only update when we know the device actually responded. Having this here could lead HE to become out to sync with the device.

Reply:
Hi @dcmeglio - sorry for the delayed reply.. yes, that makes sense. That line was actually added in the initial version sent to me so I never really looked at it carefully.

Now that I've had time to test using BHyve as a switch, I do notice that I get 2 notifications when turning the hose on.. guessing 1 of them is this and the other is when the valve actually opens.

anyway, i'll remove all of these sendEvents in this driver

fwiw - the hose is now working great with Google voice ("Ok google, turn hose on") so having the switch capability is really nice - at least for me!